Add Redis compatibility reference to the migration page (#426)
valkey-io/valkey-doc: topics/migration.md

## Why to migrate to Valkey?

> Valkey is a vendor-neutral, open-source continuation of Redis OSS.

Reviewer: Since Redis 8 with AGPL technically is open source (although the license is not permissive) so we can clarify that Valkey is using the original BSD license.

```suggestion
Valkey is a vendor-neutral, open-source continuation of Redis OSS under the original BSD license.
```

It keeps compatibility with Redis OSS 7.2 and earlier versions while adding Valkey-specific features and improvements in later releases.

## Redis compatibility and migration options

You can migrate a Redis server to Valkey.
Valkey is compatible with Redis OSS 7.2 and all earlier open-source Redis versions, as Valkey 7.2.4 is a fork of Redis OSS 7.2.4.

> Migrating from any open-source Redis version to Valkey is effectively an upgrade.

Reviewer: Technically, Redis 8 added AGPL as an option, which is officially an open-source licence, so we can formulate it differently to avoid lying.

```suggestion
Valkey is compatible with Redis OSS 7.2 and all earlier open-source Redis versions, as Valkey 7.2.4 is a fork of Redis OSS 7.2.4.
Migrating from one of these Redis versions to Valkey is effectively an upgrade.
```

> NOTE: In this guide, whenever a reference to a `redis-cli` or `valkey-cli` command is provided, the reference will only point to the Valkey version of the documentation.

> Redis Community Edition (CE) versions 7.4 and later are not open source, and their data files are not compatible with Valkey.

Reviewer: Redis 7.4 is not open source, but Redis 8 technically is, with the AGPL option. We can formulate it differently.

```suggestion
Redis Community Edition (CE) version 7.4 and later produce data files that are not compatible with Valkey.
```

It may be possible to migrate the data to Valkey from proprietary Redis versions and other Redis-like software, but it requires another method and is not covered by this document.

The following table provides migration options depending on the Redis version you run:

| Redis source          | Valkey target       | Notes |
|-----------------------|---------------------|-------|

> | OSS 2.x - 7.2.x       | Valkey 7.2 or later | Uses Redis OSS-compatible protocol, configuration, and RDB/AOF formats. |

Reviewer: It seems that our spellchecker CI job doesn't like the word "OSS-compatible". We can actually just say "compatible" here.

```suggestion
| OSS 2.x - 7.2.x       | Valkey 7.2 or later | Uses compatible protocol, configuration, and RDB/AOF formats. |
```
